lib/salemove/process_handler/pivot_process.rb

```diff
@@ -197,17 +195,39 @@ def log_processed_request(input, result)
             .merge(type: input[:type])
             .merge(PivotProcess.trace_information)
 
-          PivotProcess.logger.info "Processed request", attributes
+          @logger.info 'Processed request', attributes
         end
 
         def handle_exception(e, input)
-          PivotProcess.logger.error(e.inspect + "\n" + e.backtrace.join("\n"), PivotProcess.trace_information)
+          @logger.error(e.inspect + "\n" + e.backtrace.join("\n"), PivotProcess.trace_information)
           if @exception_notifier
             @exception_notifier.notify_or_ignore(e, cgi_data: ENV.to_hash, parameters: input)
           end
           { success: false, error: e.message }
         end
       end
+
+      class Benchmarker
+        def initialize(logger:, statsd:)
+          @logger = logger
+          @statsd = statsd
+        end
+
+        def call(input, &block)
+          type = input[:type] if input.is_a?(Hash)
+          result = nil
+
+          bm = Benchmark.measure { result = block.call }
+
+          @logger.info('Execution time', {
+            type: type, real: bm.real, user: bm.utime, system: bm.stime
+          }.merge(PivotProcess.trace_information))
+
+          @statsd.histogram('service.execution_time', bm.real)
```

Comment:
1. I don't think it should be hard-coded
2. `histogram` call doesn't include any tags, whereas smth like `application` and called `endpoint` could be really useful

Reply:
1. will make it configurable
2. there already are some global tags. will make application explicit and also add type

Reply:
bb8e360